backend/routes.py

```
@@ -1578,11 +1581,23 @@ def return_route(id):
                 }), 200
             except Exception as e:
                 db.session.rollback()
-                print(f"Database error during tool return: {str(e)}")
+                logger.error(f"Database error during tool return", exc_info=True, extra={
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)